val/src/avs_exerciser.c

```diff
@@ -218,11 +218,6 @@ val_exerciser_execute_tests(uint32_t level)
   uint32_t status, i;
   uint32_t num_instances;
 
-  if (level <= 3) {
```

Comment:
This is to cross-check that each module test runs at its applicable levels. Users can have a top-level check at the app (uefi_app/SbsaAvsMain.c). Since this file is platform/firmware specific, a check at ACS level is needed. Hence this check is required.

Reply:
Default SbsaAvsMain.c just runs exerciser tests. Without any level checks. So on L3 you get info "now we will run exerciser tests" with "sorry, but not on this level" right after it.

Reply:
Understood. My suggestion is to keep the check at both places(SbsaAvsMain.c and avs_exerciser.c). One at the app level and the other at the ACS level. Since the app is platform-specific, even if the user misses to code this check, it would be caught at the ACS level.